lib/net/ldap.rb

```diff
@@ -540,6 +540,9 @@ def initialize(args = {})
     @base = args[:base] || DefaultTreebase
     @force_no_page = args[:force_no_page] || DefaultForceNoPage
     @encryption = args[:encryption] # may be nil
+    if !@encryption.nil? and !@encryption.is_a?(Hash)
```

Comment:
Alternatively, we could also use a single type check with `@encryption.respond_to?(:[])`. I don't have a strong preference here.

@mynameisrufus why would `&&` fail here?